From 3622e1e0688a34ae78068f070cc435f944a2c807 Mon Sep 17 00:00:00 2001
From: Andreas Schneider <asn@cryptomilk.org>
Date: Mon, 15 Jan 2018 12:58:53 +0100
Subject: [PATCH 7/7] treble sepolicy tests: Workaround /sbin services labels

Example for Sony's tad_static:
/sbin/tad_static u:object_r:rootfs:s0

This should *not* be marked as system or vendor.

The entrypointpaths for tad_static found are:
    '/res(/.*)?', '/lib(/.*)?', '/sbin(/.*)?', '/init\\..*',
    '/fstab\\..*', '/ueventd\\..*', '/', '/d', '/etc',
    '/proc', '/root', '/sdcard', '/config', '/charger',
    '/bugreports', '/verity_key', '/lost\\+found',
    '/build\\.prop', '/default\\.prop', '/selinux_version',
    '/sbin/tad_static'

The MatchPathsPrefix(path, '/init') will match the '/init\\..*'
entrypoint and mark it as 'fromSystem'.

Change-Id: Ic18fb7d3c04aea672a77a209ac0bcf406ac95877
---
 tests/treble_sepolicy_tests.py | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/tests/treble_sepolicy_tests.py b/tests/treble_sepolicy_tests.py
index 58fd85b..62f1f6d 100644
--- a/tests/treble_sepolicy_tests.py
+++ b/tests/treble_sepolicy_tests.py
@@ -123,6 +123,10 @@ def GetCoreDomains():
             if (MatchPathPrefix(path, "/vendor") or
                     MatchPathPrefix(path, "/system/vendor")):
                 alldomains[d].fromVendor = True
+            # Work around to *not* mark /sbin services as system
+            if MatchPathPrefix(path, "/sbin"):
+                alldomains[d].fromSystem = False
+                alldomains[d].fromVendor = False
 
 ###
 # Add the entrypoint type and path(s) to each domain.
-- 
2.7.4

